bindings/python/pymongoarrow/lib.pyx

```diff
@@ -288,6 +287,8 @@ def process_bson_stream(bson_stream, context, arr_value_builder=None):
                     if value_t == BSON_TYPE_BINARY:
                         bson_iter_binary (&doc_iter, &subtype,
                             &val_buf_len, &val_buf)
+                        if subtype != builder.subtype:
+                            raise ValueError(f'Expected Binary subtype {builder.subtype}, got {subtype}')
```

Comment:
Interesting. Based on our current behavior I would have expected this to be an append_null() in the same way that a non-Binary value would lead to an append_null().

Reply:
Actually that brings up a good point about complex types.  We haven't added a test where you have an embedded document with missing or extra keys, what should be the behavior there?  By this same logic it seems like we should ignore extra keys and null out empty keys.

Reply:
Also if a key's value is of the wrong type, I presume that should be null.

Reply:
Right, I believe you are describing the existing behavior. Can you confirm?

Reply:
I added a confirming test for the embedded document behavior and updated this behavior to match.